File: spring-boot-oauth2-auth-server/src/main/java/com/bittokazi/oauth2/auth/server/app/models/tenant/security/UserOauth.java
package com.bittokazi.oauth2.auth.server.app.models.tenant.security;

import com.bittokazi.oauth2.auth.server.app.models.tenant.Role;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;
import org.hibernate.annotations.GenericGenerator;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.io.Serializable;
import java.util.*;
import java.util.stream.Collectors;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class UserOauth implements UserDetails, Serializable {

    private String id;

    private String email;

    private String username;

    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private String password;

    private boolean enabled;

    private Set<RoleOauth> roles = new HashSet<RoleOauth>();

    private Collection<? extends GrantedAuthority> authorities;

    private Boolean accountNonLocked = true;

    private Boolean accountNonExpired = true;

    private Boolean credentialsNonExpired = true;


    public Collection<? extends GrantedAuthority> getAuthorities() {
        if(Objects.nonNull(authorities)) return authorities;
        List<GrantedAuthority> rls = new ArrayList<GrantedAuthority>();
        for (RoleOauth role : roles) {
            rls.add(new SimpleGrantedAuthority(role.getName()));
        }
        return rls;
    }

    public String getPassword() {
        return password;
    }

    public String getUsername() {
        return username;
    }

    public boolean isAccountNonExpired() {
        return true;
    }

    public boolean isAccountNonLocked() {
        return true;
    }

    public boolean isCredentialsNonExpired() {
        return true;
    }

    public boolean isEnabled() {
        return true;
    }

}

File: spring-boot-oauth2-auth-server/src/main/kotlin/com/bittokazi/oauth2/auth/server/app/models/tenant/security/UserOauth.kt
package com.bittokazi.oauth2.auth.server.app.models.tenant.security

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import java.io.Serializable
import java.util.*

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
class UserOauth : UserDetails, Serializable {
    val id: String? = null

    var email: String? = null

    @JvmField
    var username: String? = null

    @JvmField
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    var password: String? = null

    val enabled = false

    var roles: Set<RoleOauth> = HashSet()

    @JvmField
    var authorities: Collection<GrantedAuthority>? = null

    val accountNonLocked = true

    val accountNonExpired = true

    val credentialsNonExpired = true


    override fun getAuthorities(): Collection<GrantedAuthority> {
        if (Objects.nonNull(authorities)) return authorities!!
        val rls: MutableList<GrantedAuthority> = ArrayList()
        for (role in roles) {
            rls.add(SimpleGrantedAuthority(role.name))
        }
        return rls
    }

    override fun getPassword(): String? {
        return password
    }

    override fun getUsername(): String? {
        return username
    }

    override fun isAccountNonExpired(): Boolean {
        return true
    }

    override fun isAccountNonLocked(): Boolean {
        return true
    }

    override fun isCredentialsNonExpired(): Boolean {
        return true
    }

    override fun isEnabled(): Boolean {
        return true
    }
}
